eth/backend.go

```diff
@@ -284,8 +284,10 @@ func New(ctx *node.ServiceContext, config *Config, tomoXServ *tomox.TomoX, lendi
 			return block, false, nil
 		}
 
-		eth.protocolManager.fetcher.SetSignHook(signHook)
-		eth.protocolManager.fetcher.SetAppendM2HeaderHook(appendM2HeaderHook)
+		if ctx.GetConfig().KeyStoreDir != "" {
```

Comment:
This check is irrelevant because **config.KeyStoreDir** is a parameter to let user override default value for KeyStoreDir, which is **DataDir/keystore**.